<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="wolfemd" />


<title>Genomic Prediction Analysis - Stage II of II: Cross-validation Round 1</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/darkly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">IITA_2019GS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Synopsis</a>
</li>
<li>
  <a href="IITA_PrepareTrainingData.html">Prep. Training Data</a>
</li>
<li>
  <a href="IITA_StageI_GetBLUPs.html">Get BLUPs</a>
</li>
<li>
  <a href="IITA_StageII_CheckPredictionAccuracy1.html">Cross-val. round 1</a>
</li>
<li>
  <a href="IITA_StageII_CheckPredictionAccuracy2.html">Cross-val. round 2</a>
</li>
<li>
  <a href="IITA_StageII_Predict_C4.html">Get GEBV for GS C4</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/wolfemd/IITA_2019GS">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Genomic Prediction Analysis - Stage II of II: Cross-validation Round 1</h1>
<h4 class="author">wolfemd</h4>
<h4 class="date">2019-7-27</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-02-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>IITA_2019GS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0.9000). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191121code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191121)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191121code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191121)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomwolfemdIITA2019GStree3a93083d8f612449e7d1de49321898cabf9e06f5targetblank3a93083a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/wolfemd/IITA_2019GS/tree/3a93083d8f612449e7d1de49321898cabf9e06f5" target="_blank">3a93083</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomwolfemdIITA2019GStree3a93083d8f612449e7d1de49321898cabf9e06f5targetblank3a93083a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  workflowr_log.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2019GS/blob/3a93083d8f612449e7d1de49321898cabf9e06f5/analysis/IITA_StageII_CheckPredictionAccuracy1.Rmd" target="_blank">3a93083</a>
</td>
<td>
wolfemd
</td>
<td>
2020-02-13
</td>
<td>
Add / fix navigation links and misc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2019GS/cb61b89dd126ae178789e25f69c610d135e4538d/docs/IITA_StageII_CheckPredictionAccuracy1.html" target="_blank">cb61b89</a>
</td>
<td>
wolfemd
</td>
<td>
2019-11-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2019GS/70242a63eaf90a35a333724b9c9e2c646924786e/docs/IITA_StageII_CheckPredictionAccuracy1.html" target="_blank">70242a6</a>
</td>
<td>
wolfemd
</td>
<td>
2019-11-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2019GS/dacbcf9f3dae0a2329ee0fb75cbf45f69e4770b6/docs/IITA_StageII_CheckPredictionAccuracy1.html" target="_blank">dacbcf9</a>
</td>
<td>
wolfemd
</td>
<td>
2019-11-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2019GS/43e9d5d4f6925429abc0a1e1f2994268b8473169/docs/IITA_StageII_CheckPredictionAccuracy1.html" target="_blank">43e9d5d</a>
</td>
<td>
wolfemd
</td>
<td>
2019-11-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/IITA_2019GS/a869b9e0e59a4a0af2abf5af6a0b635cb48b936f/docs/IITA_StageII_CheckPredictionAccuracy1.html" target="_blank">a869b9e</a>
</td>
<td>
wolfemd
</td>
<td>
2019-11-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/IITA_2019GS/blob/bfffb5185f22fceccc01443b45c6b4650586c2ef/analysis/IITA_StageII_CheckPredictionAccuracy1.Rmd" target="_blank">bfffb51</a>
</td>
<td>
wolfemd
</td>
<td>
2019-11-21
</td>
<td>
Publish the first set of analyses and files for IITA 2019 GS,
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="objective" class="section level1">
<h1>Objective</h1>
<p>Two training datasets versions:</p>
<ol style="list-style-type: decimal">
<li>studyYear&gt;2012</li>
<li>allYears</li>
</ol>
<p>Using newly imputed DArT+GBS sites to construct kinship.</p>
<p>Cross-validate the entire TP for Ismail’s desired traits. Isolate sets of folds within different chunks of the TP:</p>
<ul>
<li>GG, TMS13, TMS14, TMS15, TMS18 (Ikenne CET only)</li>
</ul>
<p>See code below and a diagram in the results section below that for more detail.</p>
<div class="figure">
<img src="../output/sketch_of_crossVal_scheme.png" alt="" />
<p class="caption">Cross-validation scheme</p>
</div>
</div>
<div id="make-a-kinship-matrix" class="section level1">
<h1>Make a kinship matrix</h1>
<div id="read-and-process-dosages" class="section level2">
<h2>Read and process dosages</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">library</span>(tidyverse); <span class="kw">library</span>(magrittr); <span class="kw">library</span>(furrr); <span class="kw">library</span>(data.table)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw">options</span>(<span class="dt">mc.cores=</span><span class="dv">18</span>)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="kw">plan</span>(multiprocess)</span>
<span id="cb1-4"><a href="#cb1-4"></a>snps&lt;-<span class="kw">tibble</span>(<span class="dt">Chr=</span><span class="dv">1</span><span class="op">:</span><span class="dv">18</span>) <span class="op">%&gt;%</span></span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">raw=</span><span class="kw">future_pmap</span>(.,<span class="cf">function</span>(Chr,...){</span>
<span id="cb1-6"><a href="#cb1-6"></a>      <span class="co"># </span><span class="al">NOTE</span><span class="co">: The filepath here directs to a directory on cbsurobbins.biohpc.cornell.edu</span></span>
<span id="cb1-7"><a href="#cb1-7"></a>      <span class="co"># It is too big to put on GitHub</span></span>
<span id="cb1-8"><a href="#cb1-8"></a>      <span class="co"># To do list is to figure out best-practice for sharing</span></span>
<span id="cb1-9"><a href="#cb1-9"></a>        filename&lt;-<span class="kw">paste0</span>(<span class="st">&quot;/workdir/marnin/nextgenImputation2019/ImputationStageII_71219/chr&quot;</span>,Chr,</span>
<span id="cb1-10"><a href="#cb1-10"></a>                         <span class="st">&quot;_ImputationReferencePanel_StageIIpartI_72219.raw&quot;</span>) </span>
<span id="cb1-11"><a href="#cb1-11"></a>        snps&lt;-<span class="kw">fread</span>(filename,</span>
<span id="cb1-12"><a href="#cb1-12"></a>                    <span class="dt">stringsAsFactors =</span> F) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb1-13"><a href="#cb1-13"></a><span class="st">            </span>as_tibble</span>
<span id="cb1-14"><a href="#cb1-14"></a>        <span class="kw">return</span>(snps) }))</span>
<span id="cb1-15"><a href="#cb1-15"></a><span class="co"># BLUPs pre-outlier removal</span></span>
<span id="cb1-16"><a href="#cb1-16"></a>asModelsFit&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span><span class="st">&quot;data/iita_blupsForCrossVal_72619.rds&quot;</span>)</span>
<span id="cb1-17"><a href="#cb1-17"></a>clonesWithBLUPs&lt;-asModelsFit <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">unnest</span>(blups) <span class="op">%$%</span><span class="st"> </span><span class="kw">unique</span>(GID)</span></code></pre></div>
<p>Subset to clones with BLUPs only.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>snps <span class="op">%&lt;&gt;%</span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">raw=</span><span class="kw">map</span>(raw,<span class="cf">function</span>(raw){</span>
<span id="cb2-3"><a href="#cb2-3"></a>    out&lt;-raw <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="st">      </span>as.data.frame <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="st">      </span><span class="kw">column_to_rownames</span>(<span class="dt">var =</span> <span class="st">&quot;IID&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb2-6"><a href="#cb2-6"></a><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>FID,<span class="op">-</span>PAT,<span class="op">-</span>MAT,<span class="op">-</span>SEX,<span class="op">-</span>PHENOTYPE) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb2-7"><a href="#cb2-7"></a><span class="st">      </span>as.matrix <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="st">      </span>.[<span class="kw">rownames</span>(.) <span class="op">%in%</span><span class="st"> </span>clonesWithBLUPs,];</span>
<span id="cb2-9"><a href="#cb2-9"></a>    <span class="kw">return</span>(out) }))</span></code></pre></div>
<p>Concatenate per-chromosome dosage matrices.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>snps&lt;-<span class="kw">reduce</span>(snps<span class="op">$</span>raw,cbind)</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="kw">dim</span>(snps) <span class="co"># [1]  6629 84957</span></span></code></pre></div>
<p>Subset the BLUPs to genotyped-clones only.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>asModelsFit <span class="op">%&lt;&gt;%</span></span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="st">      </span><span class="kw">mutate</span>(<span class="dt">blups=</span><span class="kw">map</span>(blups,<span class="op">~</span><span class="kw">filter</span>(.,GID <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(snps))))</span>
<span id="cb4-3"><a href="#cb4-3"></a>asModelsFit <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="st">      </span><span class="kw">mutate</span>(<span class="dt">Ntrain=</span><span class="kw">map_dbl</span>(blups,nrow)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="st">      </span><span class="kw">select</span>(Trait,Dataset,H2,Ntrain) <span class="op">%&gt;%</span><span class="st"> </span>as.data.frame</span></code></pre></div>
</div>
<div id="maf1-filter" class="section level2">
<h2>MAF&gt;1% filter</h2>
<p>Remove SNPs with minor allele frequency (MAF) less than 0.01</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>maf_filter&lt;-<span class="cf">function</span>(snps,thresh){</span>
<span id="cb5-2"><a href="#cb5-2"></a>    freq&lt;-<span class="kw">colMeans</span>(snps, <span class="dt">na.rm=</span>T)<span class="op">/</span><span class="dv">2</span>; maf&lt;-freq;</span>
<span id="cb5-3"><a href="#cb5-3"></a>    maf[<span class="kw">which</span>(maf <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>)]&lt;-<span class="dv">1</span><span class="op">-</span>maf[<span class="kw">which</span>(maf <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>)]</span>
<span id="cb5-4"><a href="#cb5-4"></a>    snps1&lt;-snps[,<span class="kw">which</span>(maf<span class="op">&gt;</span>thresh)];</span>
<span id="cb5-5"><a href="#cb5-5"></a>    <span class="kw">return</span>(snps1) }</span>
<span id="cb5-6"><a href="#cb5-6"></a>snps <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">maf_filter</span>(.,<span class="fl">0.01</span>)</span>
<span id="cb5-7"><a href="#cb5-7"></a><span class="kw">dim</span>(snps) <span class="co"># [1]  6629 82992</span></span></code></pre></div>
<p>Save filtered dosage matrix for future purposes.<br />
Note filepath is to a non-public location, for the time being.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="kw">saveRDS</span>(snps,<span class="dt">file=</span><span class="kw">paste0</span>(<span class="st">&quot;/workdir/marnin/IITA_2019GS/DosageMatrix_IITA_TrainingPop_72619.rds&quot;</span>))</span></code></pre></div>
</div>
<div id="construct-kinship" class="section level2">
<h2>Construct kinship</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="kw">library</span>(tidyverse); <span class="kw">library</span>(magrittr); <span class="kw">library</span>(rrBLUP)</span>
<span id="cb7-2"><a href="#cb7-2"></a>snps&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span><span class="kw">paste0</span>(<span class="st">&quot;/workdir/marnin/IITA_2019GS/DosageMatrix_IITA_TrainingPop_72619.rds&quot;</span>))</span>
<span id="cb7-3"><a href="#cb7-3"></a>K&lt;-rrBLUP<span class="op">::</span><span class="kw">A.mat</span>(snps<span class="dv">-1</span>)</span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="kw">saveRDS</span>(K,<span class="dt">file=</span><span class="kw">paste0</span>(<span class="st">&quot;data/&quot;</span>,</span>
<span id="cb7-5"><a href="#cb7-5"></a>                      <span class="st">&quot;Kinship_IITA_TrainingPop_72619.rds&quot;</span>))</span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="kw">rm</span>(snps); <span class="kw">gc</span>()</span></code></pre></div>
</div>
</div>
<div id="set-up-training-data" class="section level1">
<h1>Set-up training data</h1>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="kw">rm</span>(<span class="dt">list=</span><span class="kw">ls</span>()); <span class="kw">gc</span>()</span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="kw">library</span>(tidyverse); <span class="kw">library</span>(magrittr); </span>
<span id="cb8-3"><a href="#cb8-3"></a>blups&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span><span class="st">&quot;data/iita_blupsForCrossVal_72619.rds&quot;</span>)</span>
<span id="cb8-4"><a href="#cb8-4"></a>K&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span><span class="kw">paste0</span>(<span class="st">&quot;/workdir/IITA_2019GS/Kinship_IITA_TrainingPop_72619.rds&quot;</span>))</span>
<span id="cb8-5"><a href="#cb8-5"></a>blups <span class="op">%&lt;&gt;%</span></span>
<span id="cb8-6"><a href="#cb8-6"></a><span class="st">      </span><span class="kw">rename</span>(<span class="dt">trainingData=</span>blups) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb8-7"><a href="#cb8-7"></a><span class="st">      </span><span class="kw">mutate</span>(<span class="dt">trainingData=</span><span class="kw">map</span>(trainingData,<span class="op">~</span><span class="kw">filter</span>(.,GID <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(K))))</span>
<span id="cb8-8"><a href="#cb8-8"></a>tms13f&lt;-<span class="kw">rownames</span>(K) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;TMS13F|2013_&quot;</span>,.,<span class="dt">value =</span> T); <span class="kw">length</span>(tms13f) <span class="co"># 2395</span></span>
<span id="cb8-9"><a href="#cb8-9"></a>tms14f&lt;-<span class="kw">rownames</span>(K) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;TMS14F&quot;</span>,.,<span class="dt">value =</span> T); <span class="kw">length</span>(tms14f) <span class="co"># 2171</span></span>
<span id="cb8-10"><a href="#cb8-10"></a>tms15f&lt;-<span class="kw">rownames</span>(K) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;TMS15F&quot;</span>,.,<span class="dt">value =</span> T); <span class="kw">length</span>(tms15f) <span class="co"># 835</span></span>
<span id="cb8-11"><a href="#cb8-11"></a>gg&lt;-<span class="kw">setdiff</span>(<span class="kw">rownames</span>(K),<span class="kw">c</span>(tms13f,tms14f,tms15f)); <span class="kw">length</span>(gg) <span class="co"># 1228 (not strictly gg)</span></span>
<span id="cb8-12"><a href="#cb8-12"></a></span>
<span id="cb8-13"><a href="#cb8-13"></a>blups <span class="op">%&lt;&gt;%</span></span>
<span id="cb8-14"><a href="#cb8-14"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">seed_of_seeds=</span><span class="dv">1</span><span class="op">:</span><span class="kw">n</span>(),</span>
<span id="cb8-15"><a href="#cb8-15"></a>           <span class="dt">seeds=</span><span class="kw">map</span>(seed_of_seeds,<span class="cf">function</span>(seed_of_seeds,<span class="dt">reps=</span><span class="dv">5</span>){ </span>
<span id="cb8-16"><a href="#cb8-16"></a>               <span class="kw">set.seed</span>(seed_of_seeds); </span>
<span id="cb8-17"><a href="#cb8-17"></a>               outSeeds&lt;-<span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">1000</span>,<span class="dt">size =</span> reps,<span class="dt">replace =</span> F); </span>
<span id="cb8-18"><a href="#cb8-18"></a>               <span class="kw">return</span>(outSeeds) }))</span>
<span id="cb8-19"><a href="#cb8-19"></a>blups <span class="op">%&lt;&gt;%</span></span>
<span id="cb8-20"><a href="#cb8-20"></a><span class="st">      </span><span class="kw">select</span>(<span class="op">-</span>varcomp); <span class="kw">gc</span>()</span></code></pre></div>
<p>Create an directory for the output</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1"></a><span class="fu">mkdir</span> /workdir/IITA_2019GS/CrossVal_72719</span></code></pre></div>
</div>
<div id="cross-validation-function" class="section level1">
<h1>Cross-validation function</h1>
<p>The version write output to disk to save RAM</p>
<p>For each Trait-Dataset combination, run 5 reps of 5-fold cross-validation.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="co"># trainingData&lt;-blups$trainingData[[1]]; seeds&lt;-blups$seeds[[1]]; nfolds&lt;-5; reps&lt;-5;</span></span>
<span id="cb10-2"><a href="#cb10-2"></a>crossValidateFunc&lt;-<span class="cf">function</span>(Trait,Dataset,trainingData,seeds,<span class="dt">nfolds=</span><span class="dv">5</span>,<span class="dt">reps=</span><span class="dv">5</span>,<span class="dt">ncores=</span><span class="dv">40</span>,...){</span>
<span id="cb10-3"><a href="#cb10-3"></a>      trntstdata&lt;-trainingData <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-4"><a href="#cb10-4"></a><span class="st">            </span><span class="kw">filter</span>(GID <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(K))</span>
<span id="cb10-5"><a href="#cb10-5"></a>      K1&lt;-K[<span class="kw">rownames</span>(K) <span class="op">%in%</span><span class="st"> </span>trntstdata<span class="op">$</span>GID,</span>
<span id="cb10-6"><a href="#cb10-6"></a>            <span class="kw">rownames</span>(K) <span class="op">%in%</span><span class="st"> </span>trntstdata<span class="op">$</span>GID]</span>
<span id="cb10-7"><a href="#cb10-7"></a>      <span class="kw">rm</span>(K,trainingData); <span class="kw">gc</span>()</span>
<span id="cb10-8"><a href="#cb10-8"></a>      <span class="co"># seed&lt;-seeds[[1]]</span></span>
<span id="cb10-9"><a href="#cb10-9"></a>      <span class="co"># Nfolds=nfolds</span></span>
<span id="cb10-10"><a href="#cb10-10"></a>      makeFolds&lt;-<span class="cf">function</span>(<span class="dt">Nfolds=</span>nfolds,seed){</span>
<span id="cb10-11"><a href="#cb10-11"></a>            genotypes&lt;-<span class="kw">rownames</span>(K1)</span>
<span id="cb10-12"><a href="#cb10-12"></a>            </span>
<span id="cb10-13"><a href="#cb10-13"></a>            <span class="kw">set.seed</span>(seed)</span>
<span id="cb10-14"><a href="#cb10-14"></a>            seed_per_group&lt;-<span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">10000</span>,<span class="dt">size =</span> <span class="dv">4</span>,<span class="dt">replace =</span> <span class="ot">FALSE</span>)</span>
<span id="cb10-15"><a href="#cb10-15"></a>      </span>
<span id="cb10-16"><a href="#cb10-16"></a>            <span class="kw">set.seed</span>(seed_per_group[<span class="dv">1</span>])</span>
<span id="cb10-17"><a href="#cb10-17"></a>            FoldsThisRep_tms15&lt;-<span class="kw">tibble</span>(<span class="dt">CLONE=</span>genotypes[genotypes <span class="op">%in%</span><span class="st"> </span>tms15f],</span>
<span id="cb10-18"><a href="#cb10-18"></a>                                       <span class="dt">Group=</span><span class="st">&quot;TMS15F&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-19"><a href="#cb10-19"></a><span class="st">                  </span><span class="kw">mutate</span>(<span class="dt">Folds=</span><span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span>Nfolds,<span class="kw">nrow</span>(.),<span class="dt">replace=</span>T)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-20"><a href="#cb10-20"></a><span class="st">                  </span><span class="kw">arrange</span>(Folds) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-21"><a href="#cb10-21"></a><span class="st">                  </span><span class="kw">group_by</span>(Group,Folds) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-22"><a href="#cb10-22"></a><span class="st">                  </span><span class="kw">nest</span>(<span class="dt">.key =</span> Test)</span>
<span id="cb10-23"><a href="#cb10-23"></a>            <span class="kw">set.seed</span>(seed_per_group[<span class="dv">2</span>])</span>
<span id="cb10-24"><a href="#cb10-24"></a>            FoldsThisRep_tms14&lt;-<span class="kw">tibble</span>(<span class="dt">CLONE=</span>genotypes[genotypes <span class="op">%in%</span><span class="st"> </span>tms14f],</span>
<span id="cb10-25"><a href="#cb10-25"></a>                                       <span class="dt">Group=</span><span class="st">&quot;TMS14F&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-26"><a href="#cb10-26"></a><span class="st">                  </span><span class="kw">mutate</span>(<span class="dt">Folds=</span><span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span>Nfolds,<span class="kw">nrow</span>(.),<span class="dt">replace=</span>T)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-27"><a href="#cb10-27"></a><span class="st">                  </span><span class="kw">arrange</span>(Folds) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-28"><a href="#cb10-28"></a><span class="st">                  </span><span class="kw">group_by</span>(Group,Folds) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-29"><a href="#cb10-29"></a><span class="st">                  </span><span class="kw">nest</span>(<span class="dt">.key =</span> Test)</span>
<span id="cb10-30"><a href="#cb10-30"></a>            <span class="kw">set.seed</span>(seed_per_group[<span class="dv">3</span>])</span>
<span id="cb10-31"><a href="#cb10-31"></a>            FoldsThisRep_tms13&lt;-<span class="kw">tibble</span>(<span class="dt">CLONE=</span>genotypes[genotypes <span class="op">%in%</span><span class="st"> </span>tms13f],</span>
<span id="cb10-32"><a href="#cb10-32"></a>                                       <span class="dt">Group=</span><span class="st">&quot;TMS13F&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-33"><a href="#cb10-33"></a><span class="st">                  </span><span class="kw">mutate</span>(<span class="dt">Folds=</span><span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span>Nfolds,<span class="kw">nrow</span>(.),<span class="dt">replace=</span>T)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-34"><a href="#cb10-34"></a><span class="st">                  </span><span class="kw">arrange</span>(Folds) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-35"><a href="#cb10-35"></a><span class="st">                  </span><span class="kw">group_by</span>(Group,Folds) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-36"><a href="#cb10-36"></a><span class="st">                  </span><span class="kw">nest</span>(<span class="dt">.key =</span> Test)</span>
<span id="cb10-37"><a href="#cb10-37"></a>            <span class="kw">set.seed</span>(seed_per_group[<span class="dv">4</span>])</span>
<span id="cb10-38"><a href="#cb10-38"></a>            FoldsThisRep_gg&lt;-<span class="kw">tibble</span>(<span class="dt">CLONE=</span>genotypes[genotypes <span class="op">%in%</span><span class="st"> </span>gg],</span>
<span id="cb10-39"><a href="#cb10-39"></a>                                       <span class="dt">Group=</span><span class="st">&quot;GGetc&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-40"><a href="#cb10-40"></a><span class="st">                  </span><span class="kw">mutate</span>(<span class="dt">Folds=</span><span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span>Nfolds,<span class="kw">nrow</span>(.),<span class="dt">replace=</span>T)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-41"><a href="#cb10-41"></a><span class="st">                  </span><span class="kw">arrange</span>(Folds) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-42"><a href="#cb10-42"></a><span class="st">                  </span><span class="kw">group_by</span>(Group,Folds) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-43"><a href="#cb10-43"></a><span class="st">                  </span><span class="kw">nest</span>(<span class="dt">.key =</span> Test)</span>
<span id="cb10-44"><a href="#cb10-44"></a>           FoldsThisRep&lt;-<span class="kw">bind_rows</span>(FoldsThisRep_tms15,FoldsThisRep_tms14) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-45"><a href="#cb10-45"></a><span class="st">                  </span><span class="kw">bind_rows</span>(FoldsThisRep_tms13) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-46"><a href="#cb10-46"></a><span class="st">                  </span><span class="kw">bind_rows</span>(FoldsThisRep_gg) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-47"><a href="#cb10-47"></a><span class="st">                  </span><span class="kw">mutate</span>(<span class="dt">Test=</span><span class="kw">map</span>(Test,<span class="op">~</span>.<span class="op">$</span>CLONE),</span>
<span id="cb10-48"><a href="#cb10-48"></a>                         <span class="dt">Train=</span><span class="kw">map</span>(Test,<span class="op">~</span>genotypes[<span class="op">!</span>genotypes <span class="op">%in%</span><span class="st"> </span>.]))</span>
<span id="cb10-49"><a href="#cb10-49"></a>            <span class="kw">return</span>(FoldsThisRep) }</span>
<span id="cb10-50"><a href="#cb10-50"></a>      </span>
<span id="cb10-51"><a href="#cb10-51"></a>    crossval&lt;-<span class="kw">tibble</span>(<span class="dt">Rep=</span><span class="dv">1</span><span class="op">:</span>reps,<span class="dt">seed=</span><span class="kw">unlist</span>(seeds)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-52"><a href="#cb10-52"></a><span class="st">        </span><span class="kw">mutate</span>(<span class="dt">Folds=</span><span class="kw">map2</span>(Rep,seed,<span class="op">~</span><span class="kw">makeFolds</span>(<span class="dt">Nfolds=</span>nfolds,<span class="dt">seed=</span>.y))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-53"><a href="#cb10-53"></a><span class="st">        </span><span class="kw">unnest</span>()</span>
<span id="cb10-54"><a href="#cb10-54"></a></span>
<span id="cb10-55"><a href="#cb10-55"></a>    <span class="co">#Test&lt;-crossval$Test[[1]]; Train&lt;-crossval$Train[[1]]</span></span>
<span id="cb10-56"><a href="#cb10-56"></a>    crossValidate&lt;-<span class="cf">function</span>(Test,Train){ </span>
<span id="cb10-57"><a href="#cb10-57"></a>      train&lt;-Train</span>
<span id="cb10-58"><a href="#cb10-58"></a>      test&lt;-Test</span>
<span id="cb10-59"><a href="#cb10-59"></a>      trainingdata&lt;-trntstdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-60"><a href="#cb10-60"></a><span class="st">        </span><span class="kw">filter</span>(GID <span class="op">%in%</span><span class="st"> </span>train) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-61"><a href="#cb10-61"></a><span class="st">        </span><span class="kw">mutate</span>(<span class="dt">GID=</span><span class="kw">factor</span>(GID,<span class="dt">levels=</span><span class="kw">rownames</span>(K1)))</span>
<span id="cb10-62"><a href="#cb10-62"></a>       </span>
<span id="cb10-63"><a href="#cb10-63"></a>      <span class="kw">require</span>(sommer)</span>
<span id="cb10-64"><a href="#cb10-64"></a>      proctime&lt;-<span class="kw">proc.time</span>()</span>
<span id="cb10-65"><a href="#cb10-65"></a>      fit &lt;-<span class="st"> </span><span class="kw">mmer</span>(<span class="dt">fixed =</span> drgBLUP <span class="op">~</span><span class="dv">1</span>,</span>
<span id="cb10-66"><a href="#cb10-66"></a>                  <span class="dt">random =</span> <span class="op">~</span><span class="kw">vs</span>(GID,<span class="dt">Gu=</span>K1),</span>
<span id="cb10-67"><a href="#cb10-67"></a>                  <span class="dt">weights =</span> WT,</span>
<span id="cb10-68"><a href="#cb10-68"></a>                  <span class="dt">data=</span>trainingdata) </span>
<span id="cb10-69"><a href="#cb10-69"></a>      <span class="kw">proc.time</span>()<span class="op">-</span>proctime</span>
<span id="cb10-70"><a href="#cb10-70"></a>      </span>
<span id="cb10-71"><a href="#cb10-71"></a>      x&lt;-fit<span class="op">$</span>U<span class="op">$</span><span class="st">`</span><span class="dt">u:GID</span><span class="st">`</span><span class="op">$</span>drgBLUP </span>
<span id="cb10-72"><a href="#cb10-72"></a>      gebvs&lt;-<span class="kw">tibble</span>(<span class="dt">GID=</span><span class="kw">names</span>(x),</span>
<span id="cb10-73"><a href="#cb10-73"></a>                    <span class="dt">GEBV=</span><span class="kw">as.numeric</span>(x))</span>
<span id="cb10-74"><a href="#cb10-74"></a>      </span>
<span id="cb10-75"><a href="#cb10-75"></a>      accuracy&lt;-gebvs <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-76"><a href="#cb10-76"></a><span class="st">            </span><span class="kw">filter</span>(GID <span class="op">%in%</span><span class="st"> </span>test) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-77"><a href="#cb10-77"></a><span class="st">            </span><span class="kw">left_join</span>(</span>
<span id="cb10-78"><a href="#cb10-78"></a>                  trntstdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-79"><a href="#cb10-79"></a><span class="st">                        </span>dplyr<span class="op">::</span><span class="kw">select</span>(GID,BLUP) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-80"><a href="#cb10-80"></a><span class="st">                        </span><span class="kw">filter</span>(GID <span class="op">%in%</span><span class="st"> </span>test)) <span class="op">%$%</span><span class="st"> </span></span>
<span id="cb10-81"><a href="#cb10-81"></a><span class="st">            </span><span class="kw">cor</span>(GEBV,BLUP, <span class="dt">use=</span><span class="st">&#39;complete.obs&#39;</span>) </span>
<span id="cb10-82"><a href="#cb10-82"></a>    <span class="kw">return</span>(accuracy)</span>
<span id="cb10-83"><a href="#cb10-83"></a>    }</span>
<span id="cb10-84"><a href="#cb10-84"></a>    </span>
<span id="cb10-85"><a href="#cb10-85"></a>    <span class="kw">require</span>(furrr)</span>
<span id="cb10-86"><a href="#cb10-86"></a>    <span class="kw">options</span>(<span class="dt">mc.cores=</span>ncores)</span>
<span id="cb10-87"><a href="#cb10-87"></a>    <span class="kw">plan</span>(multiprocess)</span>
<span id="cb10-88"><a href="#cb10-88"></a>    crossval&lt;-crossval <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-89"><a href="#cb10-89"></a><span class="st">          </span><span class="kw">mutate</span>(<span class="dt">accuracy=</span><span class="kw">future_map2</span>(Test,Train,<span class="op">~</span><span class="kw">crossValidate</span>(<span class="dt">Test=</span>.x,<span class="dt">Train=</span>.y)))</span>
<span id="cb10-90"><a href="#cb10-90"></a>    <span class="kw">saveRDS</span>(crossval,<span class="dt">file=</span><span class="kw">paste0</span>(<span class="st">&quot;output/CrossVal_72719/&quot;</span>,</span>
<span id="cb10-91"><a href="#cb10-91"></a>                                 <span class="st">&quot;CrossVal_&quot;</span>,Trait,<span class="st">&quot;_&quot;</span>,Dataset,<span class="st">&quot;_72719.rds&quot;</span>))</span>
<span id="cb10-92"><a href="#cb10-92"></a>    <span class="kw">rm</span>(<span class="dt">list=</span><span class="kw">ls</span>()); <span class="kw">gc</span>() </span>
<span id="cb10-93"><a href="#cb10-93"></a>    }</span></code></pre></div>
</div>
<div id="run-cv-on-two-servers" class="section level1">
<h1>Run CV on two servers</h1>
<div id="cbsumm21-40" class="section level2">
<h2>cbsumm21 (40)</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>blups <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="st">      </span><span class="kw">slice</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">23</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb11-3"><a href="#cb11-3"></a><span class="st">      </span><span class="kw">mutate</span>(<span class="dt">CVaccuracy=</span><span class="kw">pmap</span>(.,crossValidateFunc))</span></code></pre></div>
</div>
<div id="cbsu30-40" class="section level2">
<h2>cbsu30 (40)</h2>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>blups <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="st">      </span><span class="kw">slice</span>(<span class="dv">24</span><span class="op">:</span><span class="dv">46</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb12-3"><a href="#cb12-3"></a><span class="st">      </span><span class="kw">mutate</span>(<span class="dt">CVaccuracy=</span><span class="kw">pmap</span>(.,crossValidateFunc))</span></code></pre></div>
</div>
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="co">#rm(list=ls());gc()</span></span>
<span id="cb13-2"><a href="#cb13-2"></a><span class="kw">library</span>(tidyverse); </span>
<span id="cb13-3"><a href="#cb13-3"></a><span class="kw">library</span>(magrittr); </span>
<span id="cb13-4"><a href="#cb13-4"></a><span class="kw">library</span>(cowplot); </span>
<span id="cb13-5"><a href="#cb13-5"></a>files&lt;-<span class="kw">list.files</span>(<span class="st">&quot;output/CrossVal_72719/&quot;</span>)</span>
<span id="cb13-6"><a href="#cb13-6"></a>pathway&lt;-<span class="st">&quot;output/CrossVal_72719/&quot;</span></span>
<span id="cb13-7"><a href="#cb13-7"></a>cv&lt;-<span class="kw">tibble</span>(<span class="dt">Files=</span>files) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb13-8"><a href="#cb13-8"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">cvResults=</span><span class="kw">map</span>(Files,<span class="op">~</span><span class="kw">readRDS</span>(<span class="kw">paste0</span>(pathway,.))))</span>
<span id="cb13-9"><a href="#cb13-9"></a>cv <span class="op">%&lt;&gt;%</span><span class="st"> </span></span>
<span id="cb13-10"><a href="#cb13-10"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Files=</span><span class="kw">gsub</span>(pathway,<span class="st">&quot;&quot;</span>,Files),</span>
<span id="cb13-11"><a href="#cb13-11"></a>         <span class="dt">Files=</span><span class="kw">gsub</span>(<span class="st">&quot;_72719.rds&quot;</span>,<span class="st">&quot;&quot;</span>,Files),</span>
<span id="cb13-12"><a href="#cb13-12"></a>         <span class="dt">Dataset=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;2013toPresent&quot;</span>,Files),<span class="st">&quot;2013toPresent&quot;</span>,<span class="st">&quot;HistoricalDataIncluded&quot;</span>),</span>
<span id="cb13-13"><a href="#cb13-13"></a>         <span class="dt">Files=</span><span class="kw">gsub</span>(<span class="st">&quot;_2013toPresent&quot;</span>,<span class="st">&quot;&quot;</span>,Files),</span>
<span id="cb13-14"><a href="#cb13-14"></a>         <span class="dt">Files=</span><span class="kw">gsub</span>(<span class="st">&quot;_HistoricalDataIncluded&quot;</span>,<span class="st">&quot;&quot;</span>,Files)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb13-15"><a href="#cb13-15"></a><span class="st">  </span><span class="kw">rename</span>(<span class="dt">Trait=</span>Files) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb13-16"><a href="#cb13-16"></a><span class="st">  </span><span class="kw">unnest</span>(<span class="dt">cols =</span> cvResults) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb13-17"><a href="#cb13-17"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Ntrain=</span><span class="kw">map_dbl</span>(Train,<span class="op">~</span><span class="kw">length</span>(.)),</span>
<span id="cb13-18"><a href="#cb13-18"></a>         <span class="dt">Ntest=</span><span class="kw">map_dbl</span>(Test,<span class="op">~</span><span class="kw">length</span>(.))) <span class="op">%&gt;%</span></span>
<span id="cb13-19"><a href="#cb13-19"></a><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span>Test,<span class="op">-</span>Train) <span class="op">%&gt;%</span></span>
<span id="cb13-20"><a href="#cb13-20"></a><span class="st">  </span><span class="kw">unnest</span>(<span class="dt">cols =</span> accuracy)</span></code></pre></div>
<div id="figure-1" class="section level2">
<h2>Figure 1</h2>
<p>This plot shows only the analyses with all historical data included. It aims to compare the accuracy within each genetic group.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="kw">library</span>(viridis)</span>
<span id="cb14-2"><a href="#cb14-2"></a>cv <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(Dataset<span class="op">==</span><span class="st">&quot;HistoricalDataIncluded&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb14-3"><a href="#cb14-3"></a><span class="st">      </span><span class="kw">ggplot</span>(.,<span class="kw">aes</span>(<span class="dt">x=</span>Trait,<span class="dt">y=</span>accuracy,<span class="dt">fill=</span>Group)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb14-4"><a href="#cb14-4"></a><span class="st">      </span><span class="kw">geom_boxplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb14-5"><a href="#cb14-5"></a><span class="st">      </span><span class="co">#facet_wrap(~Group,nrow=1) +</span></span>
<span id="cb14-6"><a href="#cb14-6"></a><span class="st">      </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>,<span class="dt">color=</span><span class="st">&#39;darkred&#39;</span>,<span class="dt">size=</span><span class="fl">1.25</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb14-7"><a href="#cb14-7"></a><span class="st">      </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb14-8"><a href="#cb14-8"></a><span class="st">      </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">90</span>,<span class="dt">face=</span><span class="st">&#39;bold&#39;</span>)) <span class="op">+</span></span>
<span id="cb14-9"><a href="#cb14-9"></a><span class="st">      </span><span class="kw">scale_fill_viridis_d</span>()</span></code></pre></div>
<p><img src="figure/IITA_StageII_CheckPredictionAccuracy1.Rmd/unnamed-chunk-14-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/wolfemd/IITA_2019GS/blob/a869b9e0e59a4a0af2abf5af6a0b635cb48b936f/docs/figure/IITA_StageII_CheckPredictionAccuracy1.Rmd/unnamed-chunk-14-1.png" target="_blank">a869b9e</a>
</td>
<td>
wolfemd
</td>
<td>
2019-11-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="figure-2" class="section level2">
<h2>Figure 2</h2>
<p>This plot shows only the analyses excluding historical data</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="kw">library</span>(viridis)</span>
<span id="cb15-2"><a href="#cb15-2"></a>cv <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(Dataset<span class="op">!=</span><span class="st">&quot;HistoricalDataIncluded&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb15-3"><a href="#cb15-3"></a><span class="st">      </span><span class="kw">ggplot</span>(.,<span class="kw">aes</span>(<span class="dt">x=</span>Trait,<span class="dt">y=</span>accuracy,<span class="dt">fill=</span>Group)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb15-4"><a href="#cb15-4"></a><span class="st">      </span><span class="kw">geom_boxplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb15-5"><a href="#cb15-5"></a><span class="st">      </span><span class="co">#facet_wrap(~Group,nrow=1) +</span></span>
<span id="cb15-6"><a href="#cb15-6"></a><span class="st">      </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>,<span class="dt">color=</span><span class="st">&#39;darkred&#39;</span>,<span class="dt">size=</span><span class="fl">1.25</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb15-7"><a href="#cb15-7"></a><span class="st">      </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb15-8"><a href="#cb15-8"></a><span class="st">      </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">90</span>,<span class="dt">face=</span><span class="st">&#39;bold&#39;</span>)) <span class="op">+</span></span>
<span id="cb15-9"><a href="#cb15-9"></a><span class="st">      </span><span class="kw">scale_fill_viridis_d</span>()</span></code></pre></div>
<p><img src="figure/IITA_StageII_CheckPredictionAccuracy1.Rmd/unnamed-chunk-15-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/wolfemd/IITA_2019GS/blob/a869b9e0e59a4a0af2abf5af6a0b635cb48b936f/docs/figure/IITA_StageII_CheckPredictionAccuracy1.Rmd/unnamed-chunk-15-1.png" target="_blank">a869b9e</a>
</td>
<td>
wolfemd
</td>
<td>
2019-11-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="figure-3" class="section level2">
<h2>Figure 3</h2>
<p>This plot aims to compare training on all historical-included to using only 2013toPresent data. Though the advantage is not huge, I see a small benefit or no cost to including historical data. Thoughts? One thing to note: You can see that the advantage of historical data is, not suprisingly, most sig. in the Genetic Gain. Phenotyping your clones continuously for 20 years works.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="kw">library</span>(viridis)</span>
<span id="cb16-2"><a href="#cb16-2"></a>cv <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb16-3"><a href="#cb16-3"></a><span class="st">      </span><span class="kw">ggplot</span>(.,<span class="kw">aes</span>(<span class="dt">x=</span>Trait,<span class="dt">y=</span>accuracy,<span class="dt">fill=</span>Dataset)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb16-4"><a href="#cb16-4"></a><span class="st">      </span><span class="kw">geom_boxplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb16-5"><a href="#cb16-5"></a><span class="st">      </span><span class="kw">facet_wrap</span>(<span class="op">~</span>Group,<span class="dt">nrow=</span><span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb16-6"><a href="#cb16-6"></a><span class="st">      </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>,<span class="dt">color=</span><span class="st">&#39;darkred&#39;</span>,<span class="dt">size=</span><span class="fl">1.25</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb16-7"><a href="#cb16-7"></a><span class="st">      </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb16-8"><a href="#cb16-8"></a><span class="st">      </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">90</span>,<span class="dt">size =</span> <span class="dv">10</span>),</span>
<span id="cb16-9"><a href="#cb16-9"></a>            <span class="dt">legend.position =</span> <span class="st">&quot;top&quot;</span>) <span class="op">+</span></span>
<span id="cb16-10"><a href="#cb16-10"></a><span class="st">      </span><span class="kw">scale_fill_viridis_d</span>()</span></code></pre></div>
<p><img src="figure/IITA_StageII_CheckPredictionAccuracy1.Rmd/unnamed-chunk-16-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/wolfemd/IITA_2019GS/blob/a869b9e0e59a4a0af2abf5af6a0b635cb48b936f/docs/figure/IITA_StageII_CheckPredictionAccuracy1.Rmd/unnamed-chunk-16-1.png" target="_blank">a869b9e</a>
</td>
<td>
wolfemd
</td>
<td>
2019-11-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="figure-4" class="section level2">
<h2>Figure 4</h2>
<p>Or again, if you don’t want to compare the genetic groups, this is just lumping everything per trait, historical data-included.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="kw">library</span>(viridis)</span>
<span id="cb17-2"><a href="#cb17-2"></a>cv <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(Dataset<span class="op">==</span><span class="st">&quot;HistoricalDataIncluded&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb17-3"><a href="#cb17-3"></a><span class="st">      </span><span class="kw">ggplot</span>(.,<span class="kw">aes</span>(<span class="dt">x=</span>Trait,<span class="dt">y=</span>accuracy,<span class="dt">fill=</span>Trait)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb17-4"><a href="#cb17-4"></a><span class="st">      </span><span class="kw">geom_boxplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb17-5"><a href="#cb17-5"></a><span class="st">      </span><span class="co">#facet_wrap(~Group,nrow=1) +</span></span>
<span id="cb17-6"><a href="#cb17-6"></a><span class="st">      </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>,<span class="dt">color=</span><span class="st">&#39;darkred&#39;</span>,<span class="dt">size=</span><span class="fl">1.25</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb17-7"><a href="#cb17-7"></a><span class="st">      </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb17-8"><a href="#cb17-8"></a><span class="st">      </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">90</span>),</span>
<span id="cb17-9"><a href="#cb17-9"></a>            <span class="dt">legend.position =</span> <span class="st">&#39;none&#39;</span>) <span class="op">+</span></span>
<span id="cb17-10"><a href="#cb17-10"></a><span class="st">      </span><span class="kw">scale_fill_viridis_d</span>()</span></code></pre></div>
<p><img src="figure/IITA_StageII_CheckPredictionAccuracy1.Rmd/unnamed-chunk-17-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/wolfemd/IITA_2019GS/blob/a869b9e0e59a4a0af2abf5af6a0b635cb48b936f/docs/figure/IITA_StageII_CheckPredictionAccuracy1.Rmd/unnamed-chunk-17-1.png" target="_blank">a869b9e</a>
</td>
<td>
wolfemd
</td>
<td>
2019-11-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="figure-5" class="section level2">
<h2>Figure 5</h2>
<p>Adjust the y-limit from Figure 4</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="kw">library</span>(viridis)</span>
<span id="cb18-2"><a href="#cb18-2"></a>cv <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(Dataset<span class="op">==</span><span class="st">&quot;HistoricalDataIncluded&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-3"><a href="#cb18-3"></a><span class="st">      </span><span class="kw">ggplot</span>(.,<span class="kw">aes</span>(<span class="dt">x=</span>Trait,<span class="dt">y=</span>accuracy,<span class="dt">fill=</span>Trait)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb18-4"><a href="#cb18-4"></a><span class="st">      </span><span class="kw">geom_boxplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb18-5"><a href="#cb18-5"></a><span class="st">      </span><span class="co">#facet_wrap(~Group,nrow=1) +</span></span>
<span id="cb18-6"><a href="#cb18-6"></a><span class="st">      </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>,<span class="dt">color=</span><span class="st">&#39;darkred&#39;</span>,<span class="dt">size=</span><span class="fl">1.25</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb18-7"><a href="#cb18-7"></a><span class="st">      </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb18-8"><a href="#cb18-8"></a><span class="st">      </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">90</span>),</span>
<span id="cb18-9"><a href="#cb18-9"></a>            <span class="dt">legend.position =</span> <span class="st">&#39;none&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">lims</span>(<span class="dt">y=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb18-10"><a href="#cb18-10"></a><span class="st">      </span><span class="kw">scale_fill_viridis_d</span>()</span></code></pre></div>
<p><img src="figure/IITA_StageII_CheckPredictionAccuracy1.Rmd/unnamed-chunk-18-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/wolfemd/IITA_2019GS/blob/a869b9e0e59a4a0af2abf5af6a0b635cb48b936f/docs/figure/IITA_StageII_CheckPredictionAccuracy1.Rmd/unnamed-chunk-18-1.png" target="_blank">a869b9e</a>
</td>
<td>
wolfemd
</td>
<td>
2019-11-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="next-step" class="section level1">
<h1>Next step</h1>
<p><a href="IITA_StageII_CheckPredictionAccuracy2.html">Stage II: Cross-validation Run 2</a></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] viridis_0.5.1     viridisLite_0.3.0 cowplot_1.0.0     magrittr_1.5     
 [5] forcats_0.4.0     stringr_1.4.0     dplyr_0.8.3       purrr_0.3.3      
 [9] readr_1.3.1       tidyr_1.0.0       tibble_2.1.3      ggplot2_3.2.1    
[13] tidyverse_1.2.1  

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.5     xfun_0.11            haven_2.2.0         
 [4] lattice_0.20-38      colorspace_1.4-1     vctrs_0.2.0         
 [7] generics_0.0.2       htmltools_0.4.0      yaml_2.2.0          
[10] rlang_0.4.1          later_1.0.0          pillar_1.4.2        
[13] withr_2.1.2          glue_1.3.1           modelr_0.1.5        
[16] readxl_1.3.1         lifecycle_0.1.0      munsell_0.5.0       
[19] gtable_0.3.0         workflowr_1.5.0.9000 cellranger_1.1.0    
[22] rvest_0.3.5          evaluate_0.14        labeling_0.3        
[25] knitr_1.26           httpuv_1.5.2         broom_0.5.2         
[28] Rcpp_1.0.3           promises_1.1.0       backports_1.1.5     
[31] scales_1.1.0         jsonlite_1.6         farver_2.0.1        
[34] fs_1.3.1             gridExtra_2.3        hms_0.5.2           
[37] digest_0.6.22        stringi_1.4.3        grid_3.6.1          
[40] rprojroot_1.3-2      cli_1.1.0            tools_3.6.1         
[43] lazyeval_0.2.2       crayon_1.3.4         whisker_0.4         
[46] pkgconfig_2.0.3      zeallot_0.1.0        xml2_1.2.2          
[49] lubridate_1.7.4      assertthat_0.2.1     rmarkdown_1.17      
[52] httr_1.4.1           rstudioapi_0.10      R6_2.4.1            
[55] nlme_3.1-142         git2r_0.26.1         compiler_3.6.1      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
